addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -1258,7 +1258,7 @@ var FieldMonetary = NumericField.extend({
         // Prepare and add the input
         var def = this._prepareInput(this.$input).appendTo(this.$el);
 
-        if (this.currency) {
+        if (this.currency && !this.nodeOptions.no_symbol) {
```

Comment:
not sure why you need this (please add an explanation in the commit message), but maybe you could simply use a fieldfloat instead?  If that does not work, then this change is ok, but it still needs a test